src/H5F.c

```diff
@@ -462,6 +462,49 @@ H5Fget_vfd_handle(hid_t file_id, hid_t fapl_id, void **file_handle /*out*/)
     FUNC_LEAVE_API(ret_value)
 } /* end H5Fget_vfd_handle() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5Fget_file_driver_handle
+ *
+ * Purpose:     Returns a pointer to the handle of the low-level file
+ *              driver.
+ *
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5Fget_file_driver_handle(hid_t file_id, void **file_handle /*out*/)
+{
+    H5F_t *                          obj;                 /* File driver handle */
+    H5VL_object_t *                  vol_obj;             /* File info */
+    H5VL_optional_args_t             vol_cb_args;         /* Arguments to VOL callback */
+    H5VL_native_file_optional_args_t file_opt_args;       /* Arguments for optional operation */
```

Comment:
You need to look at the other 'file optional' calls to see how to use these and wire up this call through the VOL.

Reply:
Since I'm simply returning the private file handle previously stored by the VFD under `obj->shared->lf`, would we have any advantage on implementing a "file optional" handler at the VFD driver to get that same data? Do non-native VFDs behave differently, somehow?